### ISSUE TITLE
MacOS + clang runtime error 

### ISSUE THREAD

**Issue by j-roberts** [BrainDynamicsUSYD/nftsim#118]
Compiling neurofield on a mac seems to have completed without error (apart from warnings like in #116), but none of the example configs run. They error at:

`Unable to find next input variable: Population 1`

The output contains the config file up to the capital P in that line (just after the coupling matrix).

clang version:
```
clang++ -v

Apple LLVM version 8.1.0 (clang-802.0.42)
Target: x86_64-apple-darwin16.7.0
```

@RomeshA provided a fix that worked:

> working using gcc 7 installed via Homebrew. I edited the makefile to swap the ‘Linux’ and ‘Darwin’ compiler flags to use their default Linux flags, and set ‘CXX := g++-7’

**Comment by benfulcher** [comment]
@RomeshA 's fix still works. `brew install gcc` (takes ~30 mins) and then pointing the MakeFile to use the newly-installed gcc instead of clang (FYI needed a `make clean` and latest version uses the line `CXX := g++-8`)

**Comment by DKPeterson** [comment]
I've just cloned the repo and make'd the sources.
Encountering the same problem as the OP, 3 years ago.
My config: macOS 10.15.5 with current gcc:

>  % g++ --version
> Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/4.2.1
> Apple clang version 11.0.3 (clang-1103.0.32.62)
> Target: x86_64-apple-darwin19.5.0
> Thread model: posix
> InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin

I edited the Makefile  (CXX := g++) and this seems to have bee "used" during make:

> % make          
> g++ -std=c++11 -MM -MP -MT obj/wave_legacy.o src/wave_legacy.cpp > dep/wave_legacy.d
> g++ -std=c++11 -MM -MP -MT obj/wave.o src/wave.cpp > dep/wave.d
> ...
> {no errors reported during make}

But when I run the demo, it gives the same error message reported by the OP:

> % ./bin/nftsim -i ./configs/eirs-corticothalamic.conf -o eirs-corticothalamic.output
> 
> Unable to find next input variable: Population 1
> From configuration file:
> Example configuration file for the Robinson et. al. corticothalamic model

and results in an empty .output file

Any further advice?

**Comment by DKPeterson** [comment]
Yep. Definitely something fubar'd in the input file scanner. If I intersperse the .conf file text lines with fake text lines, as a palate cleanser (e.g., "xxx"), like this:

> .
> .
> .
> xxx
> Coupling 1:  Map - nu: 0.001525377176
> xxx
> Coupling 2:  Map - nu: -0.003022754434
> xxx
> Coupling 3:  Map - nu: 0.0005674779589
> xxx
> Coupling 4:  Map - nu: 0.001525377176
> .
> .
> .

then it progresses through the file input and goes on to compute a solution and create a .output file. Maybe something about the EOL character(s) in macOS text files? Dunno.

**Comment by ainsliej** [comment]
Hi All, 
I am also experiencing the same thing, and @RomeshA fix doesn't seem to work for me sadly (OS 10.12.6, gcc 10.2). Adding in 'xxx' at the sections where the config is 'sticking' seems to work, but is a bit messy. Have you got any further, or still using 'xxx's @DKPeterson ? 
Really like the idea of the toolbox btw! 
Ainslie
